### PR TITLE
Performance global styles WP_Query

### LINKED ISSUE #45978: Disable post meta/term cache updating in WP_Theme_JSON_Resolver::get_user_data_from_wp_global_styles()
### Description

Initially started in https://github.com/WordPress/wordpress-develop/pull/3655.

The method currently returns a full WP_Post array while later only the ID and post_content fields are used. That's causing a lot of unnecessary overhead.

To fix this the WP_Query should be updated to disable `update_post_meta_cache` and `update_post_term_cache`. Also, any `get_post( $x, ARRAY_A );` calls should be changed to just `get_post( $x )` so that `WP_Post::to_array()` isn't called, preventing post meta lookups. Then we can create a custom array that only includes the two fields `ID` and `post_content` as proposed by @spacedmonkey.

### Step-by-step reproduction instructions

See code in https://github.com/WordPress/wordpress-develop/blob/95d11878b0d3ed89c05ae894754374620ffae1bd/src/wp-includes/class-wp-theme-json-resolver.php#L411.

### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by spacedmonkey**
## What?
Fixes #45978 

Disable priming of post meta and terms for in global styles WP_Query. This saves two database queries in FSE themes.

### CODE REVIEW

File: lib/experimental/class-wp-theme-json-resolver-gutenberg.php [hunk 1]
```diff
@@ -16,6 +16,88 @@
...
+					'field'    => 'name',
+					'terms'    => $stylesheet,
+				),
+			),
+		);
+
+		$global_style_query = new WP_Query();
+		$recent_posts       = $global_style_query->query( $args );
+		if ( count( $recent_posts ) === 1 ) {
+			$user_cpt = get_object_vars( $recent_posts[0] );
```
Comment (spacedmonkey) [inline lib/experimental/class-wp-theme-json-resolver-gutenberg.php]: The to_array method of WP_Post, does some weird stuff that loads terms and meta. We dont need this data here. So let's use `get_object_vars` instead.

**Review by felixarntz — APPROVED** [review]
@spacedmonkey Great, this looks excellent.

Should we address this in core too, or will this change here cover for it on its own?

File: lib/experimental/class-wp-theme-json-resolver-gutenberg.php [hunk 2]
```diff
@@ -16,6 +16,88 @@
...
+				),
+				true
+			);
+			if ( ! is_wp_error( $cpt_post_id ) ) {
+				$user_cpt = get_object_vars( get_post( $cpt_post_id ) );
+			}
+		}
+
+		return $user_cpt;
+	}
```
Comment (ocean90) [inline lib/experimental/class-wp-theme-json-resolver-gutenberg.php]: The diff against the version in WordPress core:

```diff
@@ -34,15 +34,17 @@
 		$user_cpt         = array();
 		$post_type_filter = 'wp_global_styles';
 		$stylesheet       = $theme->get_stylesheet();
-		$args             = array(
-			'posts_per_page'      => 1,
-			'orderby'             => 'date',
-			'order'               => 'desc',
-			'post_type'           => $post_type_filter,
-			'post_status'         => $post_status_filter,
-			'ignore_sticky_posts' => true,
-			'no_found_rows'       => true,
-			'tax_query'           => array(
+		$args = array(
+			'posts_per_page'         => 1,
+			'orderby'                => 'date',
+			'order'                  => 'desc',
+			'post_type'              => $post_type_filter,
+			'post_status'            => $post_status_filter,
+			'ignore_sticky_posts'    => true,
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'tax_query'              => array(
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'name',
@@ -54,7 +56,7 @@
 		$global_style_query = new WP_Query();
 		$recent_posts       = $global_style_query->query( $args );
 		if ( count( $recent_posts ) === 1 ) {
-			$user_cpt = get_post( $recent_posts[0], ARRAY_A );
+			$user_cpt = get_object_vars( $recent_posts[0] );
 		} elseif ( $create_post ) {
 			$cpt_post_id = wp_insert_post(
 				array(
@@ -70,7 +72,7 @@
 				true
 			);
 			if ( ! is_wp_error( $cpt_post_id ) ) {
-				$user_cpt = get_post( $cpt_post_id, ARRAY_A );
+				$user_cpt = get_object_vars( get_post( $cpt_post_id ) );
 			}
 		}
```

**Review by ocean90 — COMMENTED** [review]
Not sure if we really have to return all post fields via `get_object_vars()` or if should just limit it to the ID and post_content fields since nothing else is used. Aside from the failing tests, this is looking good.

Note that there's [a related chat on Slack](https://wordpress.slack.com/archives/C18723MQ8/p1669731634938939) on whether this is the correct file to patch or not. cc @ockham

**Comment by spacedmonkey** [comment]
> Not sure if we really have to return all post fields via get_object_vars() or if should just limit it to the ID and post_content fields since nothing else is used. Aside from the failing tests, this is looking good.

I think we fix this that in another PR. This ticket fixes the core issue, we can tweak it elsewhere.

**Review by aristath — APPROVED** [review]
Great improvement! Thank you @spacedmonkey for spotting this 👍 

The code makes sense and works as advertised.

**Pull request merged**